pypeit/datamodel.py

```diff
@@ -902,11 +902,15 @@ def _parse(cls, hdu, ext=None, transpose_table_arrays=False, debug=False,
         else:
             _hdu = hdu
 
-        _ext = np.atleast_1d(ext)
+        _ext = np.atleast_1d(np.array(ext, dtype=object))
+        str_ext = np.logical_not([isinstance(e, (int, np.integer)) for e in _ext])
```

Comment:
So are we switching from np.invert to np.logical_not. I actually don't know which one is preferred but thought they were equivalent. If so, maybe allowing both is fine. Would be good to know the answer though.

Reply:
@rcooke-ast is right that we should be switching.  `logical_not` is robust to data type:
```
$ python3
Python 3.7.6 (v3.7.6:43364a7ae0, Dec 18 2019, 14:18:50)
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy
>>> t = numpy.array([1, 0])
>>> numpy.logical_not(t)
array([False,  True])
>>> numpy.invert(t)
array([-2, -1])
>>> _t = numpy.array([True, False])
>>> numpy.logical_not(_t)
array([False,  True])
>>> numpy.invert(_t)
array([False,  True])
```

In practice, though, I don't know when or if we have used `invert` on an integer array.